src/pages/SingleProductPage.js

```diff
@@ -1,22 +1,95 @@
-import React, { useEffect } from 'react'
-import { useParams, useHistory } from 'react-router-dom'
-import { useProductsContext } from '../context/products_context'
-import { single_product_url as url } from '../utils/constants'
-import { formatPrice } from '../utils/helpers'
+import React, { useEffect } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import { useProductsContext } from '../context/products_context';
+import { single_product_url as url } from '../utils/constants';
+import { formatPrice } from '../utils/helpers';
 import {
   Loading,
   Error,
   ProductImages,
   AddToCart,
   Stars,
   PageHero,
-} from '../components'
-import styled from 'styled-components'
-import { Link } from 'react-router-dom'
-
+} from '../components';
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
 const SingleProductPage = () => {
-  return <h4>single product page</h4>
-}
+  const { id } = useParams();
+  const idNumber = Number(id); //converted to number
+  console.log(id);
```

Comment:
Currently, we're getting an error because we're trying to convert an alphanumeric id like `"64804bb82f146aaa74a85c3a"` to a number.

  
JS can't do this so it just returns `NaN` (not a number) as the value for `idNumber`  
We then try to fetch a single product from the api using the path `/api/v1/products/NaN` which will not return any results. 

We don't need to convert this to a number so we can just remove line 18 completely